aclocal.m4

```diff
@@ -1190,62 +1190,66 @@ AC_DEFUN([MP_WERROR],[
 AC_DEFUN(MP_CHECK_SQLITE_VERSION, [
 	AC_REQUIRE([MP_SQLITE3_FLAGS])
 
-	AC_MSG_CHECKING([for SQLite version in header])
-
 	mp_check_sqlite_version_cppflags_save=$CPPFLAGS
-	mp_check_sqlite_version_ldflags_save=$LDFLAGS
+	mp_check_sqlite_version_libs_save=$LDFLAGS
 	CPPFLAGS="$CPPFLAGS $CFLAGS_SQLITE3"
-	LDFLAGS="$LDFLAGS $LDFLAGS_SQLITE3"
+	LIBS="$LIBS $LIBS_SQLITE3"
 
+	AC_MSG_CHECKING([for SQLite version in header])
 	AC_CACHE_VAL([mp_cv_sqlite_version], [
-		AC_PREPROC_IFELSE(
+		AC_RUN_IFELSE(
 			[AC_LANG_SOURCE(
 				[[
 					#include <sqlite3.h>
-					#ifndef SQLITE_VERSION_NUMBER
-					#  error "SQLITE_VERSION_NUMBER undefined"
-					#else
-					int mp_sqlite_version = SQLITE_VERSION_NUMBER;
-					#endif
+					int main(int argc, char* argv[])
+					{
+						#ifndef SQLITE_VERSION_NUMBER
+						#  error "SQLITE_VERSION_NUMBER undefined"
+						#else
+							printf("%d\n", SQLITE_VERSION_NUMBER);
+						#endif
+						return 0;
+					}
 				]]
 			)],
-			[mp_cv_sqlite_version=`grep '^[[[:space:]]]*int mp_sqlite_version = [[0-9]]*;$' conftest.i | sed -E 's/[[^0-9]]*([[0-9]]+);/\1/'`],
-			[AC_MSG_ERROR("SQLITE_VERSION_NUMBER undefined or sqlite3.h not found")]
+			[mp_cv_sqlite_version=`./conftest$EXEEXT`],
+			[AC_MSG_ERROR("SQLITE_VERSION_NUMBER undefined or sqlite3.h not found")],
+			[mp_cv_sqlite_version=3022000], # this is wrong!!!
```

Comment:
This is a temporary wrong workaround to get the code to compile on Linux.
The last line is needed as a fallback for cross-compilers and is wrong.
The `AC_PREPROC_IFELSE` sounds more correct, but `grep`ping and `sed`ing the result doesn't seem to do the job in case of `gcc` compiler. Any ideas?